api/v1alpha1/accessrequest_types.go

```diff
@@ -98,6 +100,31 @@ type AccessRequest struct {
 	Status AccessRequestStatus `json:"status,omitempty"`
 }
 
+// UpdateStatus will update this AccessRequest status field based on
+// the given status and details
+func (ar *AccessRequest) UpdateStatus(status Status, details string) {
+	result := ar.Status.DeepCopy()
+	result.Status = status
+
+	// set the expiresAt only when transitioning to GrantedStatus
+	if status == GrantedStatus && result.ExpiresAt == nil {
+		expiresAt := metav1.NewTime(time.Now().Add(ar.Spec.Duration.Duration))
+		result.ExpiresAt = &expiresAt
+	}
+
+	var detailsPtr *string
+	if details != "" {
+		detailsPtr = &details
+	}
```

Comment:
Could details just be a string type? Empty strings meaning "nothing" is a pretty common convention. And it saves you from the risk of nil pointer bugs.

Reply:
When it comes to APIs design, having pointers is preferable to be able to differentiate empty strings from not provided.

Reply:
Does this API have any specific use case for "not provided"? If not, I'd favor a minimal API. You could always add the pointer later.

Reply:
> Does this API have any specific use case for "not provided"? 
Yes. this information will be exposed to the UI and I need to be able to differentiate. I think the nil pointer is mitigated by the fact that this logic is centralized and provided as a AccessRequest method.

Reply:
Sounds good!